host_core/lib/host_core/providers/provider_module.ex

```diff
@@ -184,10 +187,25 @@ defmodule HostCore.Providers.ProviderModule do
         _e -> {:error, "Received no response on health check topic from provider"}
       end
 
-    case res do
-      {:ok, _body} -> publish_health_passed(state)
-      {:error, _} -> publish_health_failed(state)
-    end
+    # Only publish health check pass/fail when state changes
+    state =
+      case res do
+        {:ok, _body} ->
+          if !state.healthy do
```

Comment:
nit: this case statement could read easier with snippets like 
```
{:ok, _body} when !state.healthy -> ...
```

not strongly opposed to the current style

Reply:
fun fact, you can't use `!` in a guard clause.

Reply:
Oh, `when not`, that's neat